cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -228,6 +230,17 @@ func ReadinessProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe
 	return createHTTPProbe(host, path, port, scheme, 0, 15, 3, 1)
 }
 
+// StartupProbe creates a Probe object with a HTTPGet handler
+func StartupProbe(host, path string, port int, scheme v1.URIScheme, timeoutForControlPlane *metav1.Duration) *v1.Probe {
+	periodSeconds, totalDuration := int32(10), constants.DefaultControlPlaneTimeout.Seconds()
+	if timeoutForControlPlane != nil {
+		totalDuration = timeoutForControlPlane.Seconds()
+	}
+	// dynamically set the value of `failureThreshold` to meet the maximum timeout for ControlPlane static Pods
+	failureThreshold := int32(math.Ceil(totalDuration / float64(periodSeconds)))
+	return createHTTPProbe(host, path, port, scheme, 15, 15, failureThreshold, periodSeconds)
```

Comment:
- we discussed that maybe initialDelaySeconds should match periodSeconds?
- how about timeoutSeconds; isn't that a bit too much for a startup probe?